src/diffusers/image_processor.py

```diff
@@ -882,3 +883,56 @@ def preprocess(
             depth = self.binarize(depth)
 
         return rgb, depth
+
+
+class IPAdapterMaskProcessor(VaeImageProcessor):
+    """
+    Image processor for IP Adapter image masks.
+
+    """
+    def __init__(self):
+        super().__init__(do_normalize=False, do_binarize=True, do_convert_grayscale=True)
+
+    def process(self, images: Union[List[PIL.Image.Image], PIL.Image.Image]) -> torch.FloatTensor:
+        """
+        Convert a PIL.Image.Image or a list of PIL.Image.Image to a torch.FloatTensor
+        """
+        if not isinstance(images, list):
```

Comment:
```suggestion
        if not isinstance(images, list):
```

do we need this? I think the `VaeImageProcessor.preprocess` is able to handle lists too

Reply:
I wonder do we need to define `preprocess()` at all here then.